### PR TITLE
Add fstat

### PR DESCRIPTION
**Pull request by thomasten**
Added support for the fstat syscall.

### CODE REVIEW

File: host/linux/syscall.c [hunk 1]
```diff
@@ -293,6 +293,27 @@ int oe_syscall_closedir_ocall(uint64_t dirp)
     return closedir((DIR*)dirp);
 }
 
+static void _stat_copy(const struct stat* st, struct oe_stat_t* buf)
+{
+    buf->st_dev = st->st_dev;
```
Comment (CodeMonkeyLeet) [inline host/linux/syscall.c:298]: It's an existing redundancy that got copied and pasted, but could you remove it?
```suggestion
```

File: host/windows/syscall.c [hunk 2]
```diff
@@ -1694,6 +1694,28 @@ int oe_syscall_closedir_ocall(uint64_t dirp)
     return ret;
 }
 
+static void _stat_copy(const struct _stat64* winstat, struct oe_stat_t* buf)
+{
+    // The macro #define st_atime st_atim.tv_sec
+    // provides backward compatibility for older version POSIX. Here we need
```
Comment (dthaler) [inline host/windows/syscall.c:1700]: Please fix grammar
```suggestion
    // provides backward compatibility for older versions of POSIX. Here we need
```

Comment (thomasten) [reply host/windows/syscall.c]: fixed and squashed

**Pull request merged**